### PR TITLE
Implement antlr parser for projections

### PR DESCRIPTION
**Pull request by amosbird**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):

- Not for changelog (changelog entry is not required)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Implement the missing antlr parser for projections as suggested by @abyss7 

Detailed description / Documentation draft:

.

### CODE REVIEW

File: src/Parsers/New/AST/AlterTableQuery.h [hunk 1]
```diff
@@ -88,19 +94,19 @@ class AlterTableClause : public INode
     private:
         enum ChildIndex : UInt8
         {
-            // ADD COLUMN or INDEX
+            // ADD COLUMN, INDEX or PROJECTION
             ELEMENT = 0,  // TableElementExpr
```
Comment (abyss7) [inline src/Parsers/New/AST/AlterTableQuery.h:98]: If you use this value for some other clause types, then please mark in comments all possible AST classes, like `Identifier`, etc.

File: src/Parsers/New/AST/AlterTableQuery.h [hunk 2]
```diff
@@ -88,19 +94,19 @@ class AlterTableClause : public INode
...
             PARTITION = 0,  // PartitionClause
             FROM = 1,       // TableIdentifier (optional)
 
-            // CLEAR
-            COLUMN = 0,  // Identifier
+            // CLEAR COLUMN, INDEX or PROJECTION
             IN = 1,      // PartitionClause
 
             // CODEC
+            COLUMN = 0,
```
Comment (abyss7) [inline src/Parsers/New/AST/AlterTableQuery.h:109]: Same here

**Review by abyss7 — CHANGES_REQUESTED** [review]
Looks like something off with an indentation in code

**Comment by qoega** [comment]
Can you commit anything to trigger build with newer master? Or I can use mergify to update branch

**Comment by amosbird** [comment]
> Can you commit anything to trigger build with newer master? Or I can use mergify to update branch

done

**Review by abyss7 — APPROVED** [review]
Please fix the conflict.

**Comment by amosbird** [comment]
> Please fix the conflict.

Done

**Pull request merged**